src/subdomains/analytics/application/services/staking-analytics.service.ts

```diff
@@ -49,13 +52,38 @@ export class StakingAnalyticsService {
       const { dateFrom, dateTo } = StakingAnalytics.getAprPeriod();
       const stakingTypes = await this.getStakingTypes();
 
+      // masternode count
+
+      // get unpaid masternodes
+      const masternodes = await getCustomRepository(MasternodeRepository).count({
+        where: { creationHash: Not(IsNull()), resignHash: IsNull() },
+      });
+
+      // calculate database balance
+      const dbBalance = await getCustomRepository(StakingRepository)
+        .createQueryBuilder('staking')
+        .leftJoin('staking.asset', 'asset')
+        .select('SUM(balance)', 'balance')
+        .where('asset.name = :name', { name: 'DFI' })
+        .getRawOne<{ balance: number }>()
+        .then((b) => b.balance);
+
+      // get unpaid masternodes
+      const unpaidMasternodes = await getCustomRepository(MasternodeRepository).count({
+        where: { creationHash: Not(IsNull()), creationFeePaid: false },
+      });
+
+      // calculate should balance (database balance - unpaid creation fees)
+
+      const tvl = dbBalance - unpaidMasternodes * Config.masternode.fee;
```

Comment:
I would ignore the unpaid masternodes for TVL.

Reply:
This code should be moved inside the for loop. We also want the TVL for yield machine.